app/src/main/java/io/appium/uiautomator2/utils/XMLHierarchy.java

```diff
@@ -29,16 +29,55 @@
 
 import javax.xml.xpath.XPath;
 import javax.xml.xpath.XPathConstants;
+import javax.xml.xpath.XPathExpression;
 import javax.xml.xpath.XPathExpressionException;
 import javax.xml.xpath.XPathFactory;
 
+import io.appium.uiautomator2.common.exceptions.InvalidSelectorException;
 import io.appium.uiautomator2.common.exceptions.UiAutomator2Exception;
 import io.appium.uiautomator2.core.AccessibilityNodeInfoDumper;
 import io.appium.uiautomator2.model.XPathFinder;
 
 
 public abstract class XMLHierarchy {
 
+
+    private static XPathExpression compileXpath(String xpathExpression) throws InvalidSelectorException {
+        XPath xpath = XPathFactory.newInstance().newXPath();
+        XPathExpression exp = null;
+        try {
+            exp = xpath.compile(xpathExpression);
+        } catch (XPathExpressionException e) {
+            throw new InvalidSelectorException("Invalid XPath expression: ", e);
+        }
+        return exp;
+    }
+
+   /* public static ArrayList<ClassInstancePair> getClassInstancePairs(XPathExpression xpathExpression, Node root) throws ElementNotFoundException {
```

Comment:
Why is this commented out? It ought to either be commented to make clear why it is here at all, or deleted full stop.